tests/preprocessing/test_generator.py

```diff
@@ -0,0 +1,133 @@
+        annotations_group = simple_generator.load_annotations_group(simple_generator.groups[0])
+
+        np.testing.assert_equal(expected_annotations_group, annotations_group)
+
+    def test_multiple(self):
+        input_annotations_group = [
+            np.array([
+                [  0,   0,  10,  10],
+                [150, 150, 350, 350]
+            ]),
+            np.array([
+                [0, 0, 1, 1]
+            ])
+        ]
+        expected_annotations_group = input_annotations_group
+
+        simple_generator = SimpleGenerator(input_annotations_group)
+        annotations_group_0 = simple_generator.load_annotations_group(simple_generator.groups[0])
+        annotations_group_1 = simple_generator.load_annotations_group(simple_generator.groups[1])
+
+        np.testing.assert_equal([expected_annotations_group[0]], annotations_group_0)
+        np.testing.assert_equal([expected_annotations_group[1]], annotations_group_1)
+
+    def test_simple_filter(self):
+        input_annotations_group = [
+            np.array([
+                [  0,   0, 10, 10],
+                [150, 150, 50, 50]
+            ]),
+        ]
+
+        expected_annotations_group = [
+            np.array([
+                [  0,   0, 10, 10],
+            ]),
+        ]
+
+        simple_generator = SimpleGenerator(input_annotations_group)
+        # expect a UserWarning
+        with pytest.warns(UserWarning):
+            annotations_group = simple_generator.load_annotations_group(simple_generator.groups[0])
+
+        np.testing.assert_equal(expected_annotations_group, annotations_group)
+
+    def test_multiple_filter(self):
+        input_annotations_group = [
+            np.array([
+                [  0,   0,  10,  10],
+                [150, 150,  50,  50],
+                [150, 150, 350, 350],
+                [350, 350, 150, 150],
+                [  1,   1,   2,   2],
+                [  2,   2,   1,   1]
+            ]),
+            np.array([
+                [0, 0, -1, -1]
+            ]),
+            np.array([
+                [-10, -10,    0,    0],
+                [-10, -10, -100, -100],
+                [ 10,  10,  100,  100]
+            ]),
+        ]
+
+        expected_annotations_group = [
+            np.array([
+                [  0,   0,  10,  10],
+                [150, 150, 350, 350],
+                [  1,   1,   2,   2]
+            ]),
+            np.zeros((0, 4)),
+            np.array([
+                [10, 10, 100, 100]
+            ])
+        ]
+
+        simple_generator = SimpleGenerator(input_annotations_group)
+        # expect a UserWarning
+        with pytest.warns(UserWarning):
+            annotations_group_0 = simple_generator.load_annotations_group(simple_generator.groups[0])
+            annotations_group_1 = simple_generator.load_annotations_group(simple_generator.groups[1])
+            annotations_group_2 = simple_generator.load_annotations_group(simple_generator.groups[2])
```

Comment:
Each of these statements should generate a warning right? The way it's written now I think the test only checks that one of them generates a warning.

Reply:
Good point, that's true.